src/btree/bt_delete.c

```diff
@@ -204,10 +204,24 @@ __wt_delete_page_rollback(WT_SESSION_IMPL *session, WT_REF *ref)
 			 */
 			__wt_free(session, ref->page_del->update_list);
 			__wt_free(session, ref->page_del);
-			goto done;
+			return;
 		}
+		/*
+		 * We wait for the change in page state, yield before
+		 * retrying, and if we've yielded enough times, start
+		 * sleeping so we don't burn CPU to no purpose.
+		 */
+		if (++yield_count < WT_THOUSAND) {
+			__wt_yield();
+			continue;
+		}
+
+		sleep_count = WT_MIN(sleep_count + WT_THOUSAND, 10000);
```

Comment:
maybe 10000 should be 10 * WT_THOUSAND, it is a little easier to read than counting zeroes. But I do not feel strongly about it.